client/pkg/transport/listener_test.go

```diff
@@ -573,3 +578,140 @@ func TestSocktOptsEmpty(t *testing.T) {
 		}
 	}
 }
+
+// TestNewListenerWithACRLFile tests when a revocation list is present.
+func TestNewListenerWithACRLFile(t *testing.T) {
+	clientTLSInfo, err := createSelfCertEx(t, "127.0.0.1", x509.ExtKeyUsageClientAuth)
+	if err != nil {
+		t.Fatalf("unable to create client cert: %v", err)
+	}
+	loaded, err := os.ReadFile(clientTLSInfo.CertFile)
+	if err != nil {
+		t.Fatalf("unable to read client cert: %v", err)
+	}
+	block, _ := pem.Decode(loaded)
+	clientCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("unable to parse client cert: %v", err)
+	}
+
+	tests := map[string]struct {
+		expectHandshakeError   bool
+		revokedCertificates    []pkix.RevokedCertificate
+		revocationListContents []byte
+	}{
+		"empty revocation list": {
+			expectHandshakeError: false,
+		},
+		"client cert is revoked": {
+			expectHandshakeError: true,
+			revokedCertificates: []pkix.RevokedCertificate{
+				{
+					SerialNumber:   clientCert.SerialNumber,
+					RevocationTime: time.Now(),
+				},
+			},
+		},
+		"invalid CRL file content": {
+			expectHandshakeError:   true,
+			revocationListContents: []byte("@invalidcontent"),
+		},
```

Comment:
Hey @ahrtr, in this case, I'm testing that the contents of the CRL file are valid. However, this is similar to the test case in which I was testing that the file existed (trying to have exhaustive coverage of the implementation code). However, given that we opted to leave out that test case, I'm curious if you think this is worth keeping or not (it does make the test code a bit more complex). See https://github.com/etcd-io/etcd/blob/1a6f87e3d022331f7ee9287ab4e7ef76b5a1b5b4/client/pkg/transport/listener_test.go#L651-L667

Reply:
Since it doesn't complicate the test case. It isn't a big deal. So let's keep it as it's.